api/enki.py

```diff
@@ -0,0 +1,542 @@
+            licenses_available = int(licenses_available),
+            patrons_in_hold_queue = int(hold)
+        )
+
+        metadata.circulation = circulationdata
+        return metadata
+
+
+    def process_one(self, element):
+        if self.include_bibliographic:
+            bibliographic = self.extract_bibliographic(element)
+        else:
+            bibliographic = None
+
+        availability = None
+        if self.include_availability:
+            availability = bibliographic.circulation
+
+        return bibliographic, availability
+
+class EnkiImport(CollectionMonitor):
+    """Import content from Enki that we don't yet have in our collection
+    """
+    SERVICE_NAME = "Enki Circulation Monitor"
+    INTERVAL_SECONDS = 500
+    PROTOCOL = EnkiAPI.ENKI_EXTERNAL
+    DEFAULT_BATCH_SIZE = 100 
+    FIVE_MINUTES = datetime.timedelta(minutes=5)
+    
+    def __init__(self, _db, collection, api_class=EnkiAPI):
+        """Constructor."""
+        super(EnkiImport, self).__init__(_db, collection)
+        self.api = api_class(_db, collection)
+        self.analytics = Analytics(_db)
+        self.bibliographic_coverage_provider = (
+            EnkiBibliographicCoverageProvider(_db, collection, api_class=self.api)
+        )
+
+    def recently_changed_ids(self, start, cutoff):
+        return self.api.recently_changed_ids(start, cutoff)
+
+    def run_once(self, start, cutoff):
+        # Give us five minutes of overlap because it's very important
+        # we don't miss anything.
+        since = start-self.FIVE_MINUTES
+        id_start = 0
+        while True:
+            availability = self.api.availability(since=since, strt=id_start, qty=self.DEFAULT_BATCH_SIZE)
+            if availability.status_code != 200:
+                self.log.error(
+                    "Could not contact Enki server for content availability. Status: %d",
+                    availability.status_code
+                )
+            content = availability.content
+            count = 0
+            for bibliographic, circulation in BibliographicParser().process_all(content):
+                self.process_book(bibliographic, circulation)
+                count += 1
+            if count == 0:
+                break
+            self._db.commit()
+            id_start += self.DEFAULT_BATCH_SIZE
+
+    def process_book(self, bibliographic, availability):
+        license_pool, new_license_pool = availability.license_pool(self._db, self.collection)
+        edition, new_edition = bibliographic.edition(self._db)
+        license_pool.edition = edition
+        policy = ReplacementPolicy(
+            identifiers=False,
+            subjects=True,
+            contributions=True,
+            formats=True,
+        )
+        availability.apply(
+            self._db,
+            license_pool.collection,
+            replace=policy,
+        )
+        if new_edition:
+            bibliographic.apply(edition, self.collection, replace=policy)
```

Comment:
When there are multiple libraries on a circ manager, you can configure which libraries have access to which collections. So you would configure the Enki collection's settings first, and then configure which libraries have access to it. You should probably use "for library in self.collection.libraries" here too, even if you don't have any Enki collections with multiple libraries yet.